pkg/util/sysprep/secret.go

```diff
@@ -0,0 +1,116 @@
+// © Broadcom. All Rights Reserved.
+// The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+// SPDX-License-Identifier: Apache-2.0
+
+package sysprep
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/vmware-tanzu/vm-operator/api/v1alpha5/sysprep"
+	"github.com/vmware-tanzu/vm-operator/pkg/util"
+	"github.com/vmware-tanzu/vm-operator/pkg/util/ptr"
+)
+
+type SecretData struct {
+	ProductID, Password, DomainPassword, ScriptText string
+}
+
+func getSysprepSecretsImpl(
+	ctx context.Context,
+	k8sClient ctrlclient.Client,
+	secretNamespace string,
+	in *sysprep.Sysprep,
+	setData bool) (SecretData, map[string]*corev1.Secret, error) {
+
+	secretData := SecretData{}
+	secrets := map[string]*corev1.Secret{}
+
+	getSecret := func(name, key string, outData *string) error {
+		// Our ctrl-runtime client does not cache secrets, so avoid repeated
+		// fetches of the same secret.
+		secret, ok := secrets[name]
+		if !ok {
+			var err error
+			secret, err = util.GetSecretResource(ctx, k8sClient, secretNamespace, name)
```

Comment:
Consider adding a quick sanity check here to skip if the secret name is empty (in case it's set to an empty value in the API)?